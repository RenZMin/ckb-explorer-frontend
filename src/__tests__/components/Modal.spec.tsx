import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import OutsideClickHandler from 'react-outside-click-handler'
import Modal from '../../components/Modal'

describe('Modal Component', () => {
  let component: ReactElement
  let component1: ReactElement

  beforeAll(() => {
    component = <Modal children={<div className="modal__child">children</div>} isShow />
    component1 = <Modal children={<div className="modal__child">children</div>} isShow={false} />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    let wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find('.modal__child')).toHaveLength(1)
    expect(wrapper.find(OutsideClickHandler)).toHaveLength(1)
    expect(wrapper.find('.modal__child').text()).toBe('children')

    wrapper = shallow(component1)
    expect(wrapper.find('.modal__child')).toHaveLength(0)
    expect(wrapper.find(OutsideClickHandler)).toHaveLength(0)
  })
})
