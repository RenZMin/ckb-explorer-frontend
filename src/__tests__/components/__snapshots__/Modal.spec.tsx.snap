// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component shallow renders 1`] = `
<div
  className="sc-bdnxRM bOvrdL"
>
  <div>
    <div
      className="modal__child"
    >
      children
    </div>
  </div>
</div>
`;
